cli/src/main/kotlin/tools/samt/cli/OutputWriter.kt

```diff
@@ -0,0 +1,40 @@
+package tools.samt.cli
+
+import tools.samt.api.plugin.CodegenFile
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.InvalidPathException
+import java.nio.file.Path
+import kotlin.io.path.*
+
+internal object OutputWriter {
+    @Throws(IOException::class)
+    fun write(outputDirectory: Path, files: List<CodegenFile>) {
+        if (!outputDirectory.exists()) {
+            try {
+                Files.createDirectories(outputDirectory)
```

Comment:
why go through the Files class here but use an extension function further down

Reply:
The extension to create all parent directories will be introduced k. Kotlin 1.9

Reply:
I'm actually blind, you are correct 🙈